food.cpp

```diff
@@ -33,6 +34,14 @@ bool operator==(const food& lhs, const food& rhs) noexcept
             && lhs.get_y() == rhs.get_y() ;
 }
 
+void food::place_randomly(std::mt19937 &rng, const double min_x, const double max_x, const double min_y, const double max_y){

```

Comment:
We could even give this function two `coordinate`  arguments
```suggestion
void food::place_randomly(std::mt19937 &rng, const coordinate& top_right, const coordinate& bottom_left){

```
And then deduce the `x`s and `y`s from those in the body of the fucntion